src/Security/Factory/OidcFactory.php

```diff
@@ -56,6 +61,20 @@ public function createAuthenticator(
         ->addArgument($config['user_identifier_property'])
         ->addArgument($config['enable_remember_me']);
 
+    $logoutListenerId = sprintf('security.logout.listener.default.%s', $firewallName);
+
+    // Check if "logout" config is specified in the firewall and "enable_end_session_listener" is set to true
+    if ($config['enable_end_session_listener'] && $container->hasDefinition($logoutListenerId)) {
+      $endSessionListenerId = sprintf('%s.%s', DrensoOidcExtension::END_SESSION_LISTENER_ID, $firewallName);
+
+      $container->setDefinition($endSessionListenerId, new Definition(OidcEndSessionSubscriber::class))
+          ->addArgument($clientReference)
+          ->addArgument(new Reference('security.http_utils'))
+          ->addArgument($container->getDefinition($logoutListenerId)->getArgument(1)) // Get the configured logout target path
+          ->addTag('kernel.event_subscriber')
```

Comment:
This needs to be updated, it should only use the firewall event dispatcher to prevent the listener to be called from other firewalls:

```suggestion
          ->addTag('kernel.event_subscriber', ['dispatcher' => 'security.event_dispatcher.' . $firewallName]);
```

Reply:
Great suggestion, did not know this can be done this way.